client/src/components/chat/chatTab.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Created by einavcarmon on 19/06/2017.
+ */
+
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+
+import TextSubmitter from './textSubmitter';
+import Message from './message';
+import { getChat } from '../../api';
+
+import Strings from '../../strings';
+
+
+export default class chatTab extends Component {
+
+    constructor (props) {
+        super(props);
+
+        this.state = {
+            chatId : props.chatId,
+            messages: [],
+            client: props.client,
+            clientId: '',
+            isSuperuser: props.isSuperuser
+        }
+
+        this.socket = props.socket;
+
+        let self = this;
+
+        getChat(this.state.chatId).then(function (chat) {
+            self.setState({messages:chat.chat_messages});
+        })
+    }
+
+    componentWillMount() {
+        let self = this;
+        this.socket.emit(`client:connected`, { chatId : this.state.chatId , client: this.state.client, clientId:this.state.user && this.state.userId });
```

Comment:
yea probably. We might have to hack the module system (I'm thinking symbolic links again) but it should work